packages/firebase_dynamic_links/lib/src/firebase_dynamic_links.dart

```diff
@@ -50,6 +61,33 @@ class FirebaseDynamicLinks {
       iosData,
     );
   }
+
+  /// Configures onLink listeners: it has two methods for success and failure.
+  void onLink({
+    OnLinkSuccessCallback onSuccess,
+    OnLinkErrorCallback onError,
+  }) {
+    _onLinkSuccess = onSuccess;
+    _onLinkError = onError;
+    channel.setMethodCallHandler(_handleMethod);
+  }
+
+  Future<dynamic> _handleMethod(MethodCall call) async {
+    switch (call.method) {
+      case "onLinkSuccess":
+        final Map<dynamic, dynamic> data =
+            call.arguments.cast<dynamic, dynamic>();
+        final PendingDynamicLinkData linkData =
+            getPendingDynamicLinkDataFromMap(data);
+        return _onLinkSuccess(linkData);
```

Comment:
Wouldn't this cause a `NullPointerException` if `onLink` was not called.

Reply:
If `onLink` is not called, then `_handleMethod` won't be call either, as long as `channel.setMethodCallHandler` wouldn't have been called.

Am I wrong?

Reply:
You're right, good point. I missed where you called `channel.setMethodCallHandler`. However, what happens if a user passes `null` to `onLink`? Can a user unregister from `onLink` callbacks or is this bad practice?

Reply:
Well, I think that's a bad practice and we could check that, but that problem exists in almost every plugin where we depend on the developers to follow the documentation and they do it wrong:

https://github.com/flutter/plugins/blob/master/packages/firebase_messaging/lib/firebase_messaging.dart